gen_codecs.go

```diff
@@ -107,30 +128,31 @@ func NewUnionCodecGenerator(codecFromIndex []*Codec) (*CodecGenerator, error) {
 
 	gen := &CodecGenerator{
 		getImports:               func() []string { return append([]string{"fmt"}, realCodec.generator.getImports()...) },
-		genNativeTypeNameSrc:     func() string { return realCodec.generator.genNativeTypeNameSrc() },
+		genNativeTypeNameSrc:     func() string { return realCodec.generator.genNativeTypeNamePtrSrc() },
 		genNativeDefaultValueSrc: func() string { return realCodec.generator.genNativeDefaultValueSrc() },
+		genNativeTypeNamePtrSrc: func() string { return realCodec.generator.genNativeTypeNamePtrSrc() },
 	}
 
 	gen.genDecodeInstanceSrc = func() string {
 		var w bytes.Buffer
 
-		w.WriteString(fmt.Sprintf("func(buf []byte) (%s, []byte, error) {\n", gen.genNativeTypeNameSrc()))
+		w.WriteString(fmt.Sprintf("func(buf []byte) (%s, []byte, error) {\n", gen.genNativeTypeNamePtrSrc()))
 		w.WriteString("tmpBuf := buf\n")
 		w.WriteString("idx, tmpBuf, err := goavro.LongNativeFromBinary(tmpBuf)\n")
-		w.WriteString(fmt.Sprintf("if err != nil { return %s, buf, err }\n", gen.genNativeDefaultValueSrc()))
+		w.WriteString(fmt.Sprintf("if err != nil { return %s, buf, err }\n", "nil"))
```

Comment:
It seems like `Sprintf` can be removed.
```suggestion
		w.WriteString("if err != nil { return nil, buf, err }\n")
```

Reply:
Yea, just me being lazy :)

I'll make the changes

Reply:
done